### ISSUE TITLE
Add-lines patcher can not be converted from .yaml to .php

### ISSUE THREAD

**Issue by alexander-schranz** [symfony/flex#984]
Stumbled over that `add-lines` implemented in https://github.com/symfony/flex/pull/975 is now used not only for `phpunit.xml.dist` files or `.js`, ... files but also for config recipes files.

Sadly used for `config/packages/..` or `config/routes/..` I personally would avoid using that and have another mechanism there.

Maybe merged `.yaml` or even own `notifier_...yaml` would be better there.

This kind of configurations can currently this way not be converted to `PHP` based recipes. And I don't see a way for do it, without somebody need todo this manually for every recipe.

This recipe patches will so fail on PHP based config recipes repo currently: https://github.com/schranz-php-recipes/symfony-recipes-php/pull/44/files#

**Comment by nicolas-grekas** [comment]
Of course. That's why we stick to yaml files. php config files are not something we support here.

**Comment by stof** [comment]
@alexander-schranz if you make an unofficial recipes server using the PHP config files, it means your builder also need to convert those `add-lines` patchers to patch PHP files instead (find which other recipe defines the original file, apply the patch on the original file, convert the patched file, and then make a diff between the converted patched and the converted original file to find out how the patch should look like for the converted recipe). But that's an issue for your own builder, not for this repo.

**Comment by alexander-schranz** [comment]
@stof the target of the project was never to have unofficial recipe server for PHP configs ;) it's more a prototype to official support PHP configs at some time in the official flex recipes here, without the need to maintain yaml and php configs inside them. So that a job can use that kind of config converter to provide via official flex support for yaml and php configs.

With the issue I wanted to make aware that moving recipes in direction of using `add-lines` for config files it will make a step backwards for have support for PHP based configs in its full potential. If the core team of symfony says they never try to plan to have support for php config in flex recipes I'm more to way of closing the project. As in the past I thought there where a direction with @nyholm config builders, remove xml and yaml config from core and some presentation I think from @chalasr if I remember correctly that there is more a direction from go away from Yaml instead of stay with it.

**Comment by stof** [comment]
We added those usages of `add-files` because we had many cases where it was necessary to edit an existing file to account for the new package instead of simply adding separate files.
In the early days of recipes where we were mostly managing Symfony DI config files, this was not an issue (as Symfony DI supports merging any number of config files together to produce the final config). But we used recipes to configure more things to improve DX, for places where merging files is not an option (JS files, the Twig base template, etc).

However, I don't think we have cases using `add-lines` to change DI config files in any official recipes (we rely on config merging instead, which is well defined). For contrib recipes, it looks like there is a few such usages (never in PRs in which the core team was involved in review AFAICT), but I find them weird. Maybe we should restrict this case (changing the contrib recipes that rely on it, and blocking such thng in the validator)
